src/table/__tests__/TableWrapper.spec.jsx

```diff
@@ -0,0 +1,79 @@
+import '../../../test-setup';
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import TableWrapper from '../TableWrapper';
+import * as TableBodyWrapper from '../TableBodyWrapper';
+import * as TableHeadWrapper from '../TableHeadWrapper';
+
+describe('<TableWrapper />', () => {
+  const sandbox = sinon.createSandbox();
+  let tableData;
+  let setPageInfo;
+  let constraints;
+  let rowsPerPage;
+
+  beforeEach(() => {
+    sandbox.replace(TableBodyWrapper, 'default', () => <tbody />);
+    sandbox.replace(TableHeadWrapper, 'default', () => <thead />);
+
+    tableData = {
+      size: { qcy: 200 },
+      rows: [{ qText: '1' }],
+    };
+    setPageInfo = sinon.spy();
+    constraints = {};
+    rowsPerPage = 100;
+  });
+
+  afterEach(() => {
+    sandbox.verifyAndRestore();
+    sandbox.resetHistory();
+  });
+
+  it('should render table', () => {
+    mockContent();
+
+    const { queryByLabelText, queryByText } = render(
+      <TableWrapper tableData={tableData} setPageInfo={setPageInfo} constraints={constraints} />
+    );
+
+    expect(queryByLabelText('sticky table')).to.be.visible;
+    expect(queryByText(`1-${rowsPerPage} of ${tableData.size.qcy}`)).to.be.visible;
+    expect(queryByText(rowsPerPage)).to.be.visible;
+  });
+  it('should call setPageInfo when clicking next page button', async () => {
+    const { findByTitle, findByText } = render(
+      <TableWrapper tableData={tableData} setPageInfo={setPageInfo} constraints={constraints} />
+    );
+    fireEvent.click(await findByTitle('Next page'));
+
+    expect(setPageInfo).to.have.been.calledWith({ top: rowsPerPage, height: rowsPerPage });
+    expect(await findByText(`101-200 of ${tableData.size.qcy}`)).to.be.visible;
+  });
+  it('should change back to first page when not on first page and no rows', async () => {
+    tableData.rows = [];
+
+    const { findByTitle } = render(
+      <TableWrapper tableData={tableData} setPageInfo={setPageInfo} constraints={constraints} />
+    );
+    fireEvent.click(await findByTitle('Next page'));
+
+    // Called when pressing the button
+    expect(setPageInfo).to.have.been.calledWith({ top: rowsPerPage, height: rowsPerPage });
```

Comment:
Interesting that this is still called, should the button not be inactive?

Reply:
Yeah this pattern is weird. i just looked at it. the pagination does not look at `rows`, only `size.qcy`. I think I should do a somewhat realistic flow here

Reply:
Actually, I think it is fine. It is a hacky way to simulate the real use case: that you confirm selections on any page past the first page and you don't select more rows than rows per page. The data paging then only fetches rows 101-200 or something when there are less than 100 rows, which returns 0 rows.